homeassistant/components/roborock/roborock_storage.py

```diff
@@ -0,0 +1,152 @@
+"""Roborock storage."""
+
+import dataclasses
+import logging
+import os
```

Comment:
Prefer `pathlib` for all the filename and file/directory manipulation. It will make some of the path concatenation and read/write operations a little less code.

Reply:
I haven't used pathlib before, but I think I did it all correctly.

Although one test is failing now, perhaps i'm patching incorrectly. I will have to look later.